### PR TITLE
Enable cleanup for /var/log/messages on Photon for vSphere Provider

### LINKED ISSUE #1647: Enable cleanup for /var/log/messages on Photon for vSphere Provider
Enable cleanup for /var/log/messages on Photon for vSphere Provider

## What steps did you take and what happened?

- Create cluster and let it run for few days.
- Run applications which pushes output to /var/log/messages (like GPU operators).
- Since no log cleanup or rotation is enabled for /var/log/messages, disk will get filled up and can lead to disk pressure

## What did you expect to happen?

/var/log/messages be rotated or cleaned up on regular interval

---

/kind bug

### PR DESCRIPTION
**Pull request by sriramandev**
## Change description

This PR adds logrotate binary and corresponding configurations to enable log rotation and cleanup for /var/log/messages in Photon for vSphere Provider

## Related issues

- Fixes #1647

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
This looks reasonable to me. Could you squash the commits?

**Comment by sriramandev** [comment]
> This looks reasonable to me. Could you squash the commits?

Squashed the commits.

File: images/capi/ansible/roles/sysprep/defaults/main.yml [hunk 1]
```diff
@@ -16,3 +16,4 @@ extra_repos: ""
 pip_conf_file: ""
 remove_extra_repos: false
 flatcar_disable_autologin: false
+journal_log_max_size: 1G
```
Comment (mboersma) [inline images/capi/ansible/roles/sysprep/defaults/main.yml:19]: Where is this used, as opposed to the `var_log_messages_maxsize` value above?

File: images/capi/ansible/roles/providers/defaults/main.yml [hunk 2]
```diff
@@ -15,3 +15,4 @@
 networkd_dispatcher_download_url: https://gitlab.com/craftyguy/networkd-dispatcher/-/archive/2.1/networkd-dispatcher-2.1.tar.gz
 packer_builder_type: ""
 build_target: virt
+var_log_messages_maxsize: 100M
```
Comment (mboersma) [inline images/capi/ansible/roles/providers/defaults/main.yml:18]: Should we make this configurable?

File: images/capi/ansible/roles/sysprep/tasks/main.yml [hunk 3]
```diff
@@ -260,4 +260,4 @@
   ansible.builtin.systemd:
     name: ssh
     enabled: true
-  when: ansible_os_family == "Debian"
+  when: ansible_os_family == "Debian"
```
Comment (mboersma) [inline images/capi/ansible/roles/sysprep/tasks/main.yml:263]: This is a nitpick, but we prefer all files in the repo to end with a newline.

File: images/capi/ansible/roles/sysprep/defaults/main.yml (hunk 1 above)
Comment (sriramandev) [reply images/capi/ansible/roles/sysprep/defaults/main.yml:19]: This shouldn't be there.... its part of next fix which I want to get in...

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**